src/pages/token/[tokenId].tsx

```diff
@@ -0,0 +1,227 @@
+import type { GetServerSideProps } from 'next';
+import { apiClients, createConfig } from '@/common/api/client';
+import { DEFAULT_MAINNET_SERVER } from '@/common/constants';
+import { Configuration as TokenMetadataApiConfiguration } from '@hirosystems/token-metadata-api-client/dist/configuration';
+
+import { TokenPage } from '@/app/token/[tokenId]';
+
+interface TokenLinks {
+  websites: string[];
+  blockchain: string[];
+  chat: string[];
+  forums: string[];
+  announcements: string[];
+  repos: string[];
+  social: string[];
+}
+
+export interface TokenInfoProps {
+  basic: {
+    name: string;
+    symbol: string;
+    totalSupply: number | null;
+  };
+  extended: {
+    categories: string[];
+
+    links: TokenLinks;
+    circulatingSupply: number | null;
+    fullyDilutedValuation: number | null;
+    tvl: number | null;
+
+    currentPrice: number | null;
+    priceChangePercentage24h: number | null;
+    currentPriceInBtc: number | null;
+    priceInBtcChangePercentage24h: number | null;
+
+    marketCap: number | null;
+
+    tradingVolume24h: number | null;
+    tradingVolumeChangePercentage24h: number | null;
+
+    developerData: {
+      forks: number | null;
+      stars: number | null;
+      subscribers: number | null;
+      total_issues: number | null;
+      closed_issues: number | null;
+      pull_requests_merged: number | null;
+      pull_request_contributors: number | null;
+      code_additions_deletions_4_weeks: {
+        additions: number | null;
+        deletions: number | null;
+      } | null;
+      commit_count_4_weeks: number | null;
+      last_4_weeks_commit_activity_series: [] | null;
+    };
+  };
+}
+
+async function searchCoinGeckoTokens(tokenSymbol: string) {
+  try {
+    const tokenSearchResponse = await (
+      await fetch(`https://api.coingecko.com/api/v3/search?query=${tokenSymbol}`)
+    ).json();
+    const token = (tokenSearchResponse?.coins || []).find(
+      (coin: { symbol: string; id: string }) =>
+        coin.symbol?.toLowerCase() === tokenSymbol.toLowerCase()
+    );
+    if (!token?.id) {
+      console.error("couldn't find token in Coingecko");
+    }
+    return token;
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+async function getTokenFromCoinGecko(tokenId: string) {
+  try {
+    const tokenInfoResponse = await (
+      await fetch(`https://api.coingecko.com/api/v3/coins/${tokenId}?localization=false`)
+    ).json();
+    return tokenInfoResponse;
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+export const getServerSideProps: GetServerSideProps = async ({ params, query }) => {
```

Comment:
Is there any specific reason to fetch server-side ?
I think we could face some rate-limit issues from coingecko if many users are loading the token page.
Since there is no caching by default for request from node, is there anything setup on the infra side for caching ?

Reply:
We'll have caching very soon (waiting for [this](https://github.com/hirosystems/devops/issues/1264))

Reply:
Perfect 👍